sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -182,6 +183,10 @@ case class DataSource(
       throw new AnalysisException(
         s"Unable to infer schema for $format. It must be specified manually.")
     }
+
+    SchemaUtils.checkSchemaColumnNameDuplication(
+      dataSchema, "the datasource", sparkSession.sessionState.conf.caseSensitiveAnalysis)
```

Comment:
@maropu Could you submit a new PR to 2.2 now? We should fix this issue in 2.2 now. 

Just opened the JIRA: https://issues.apache.org/jira/browse/SPARK-21144